src/util/bip32.rs

```diff
@@ -169,8 +169,11 @@ impl From<ChildNumber> for u32 {
 impl fmt::Display for ChildNumber {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match *self {
-            ChildNumber::Hardened { index } => write!(f, "{}'", index),
-            ChildNumber::Normal { index } => write!(f, "{}", index),
+            ChildNumber::Hardened { index } => {
+                fmt::Display::fmt(&index, f)?;
+                f.write_str(if f.alternate() { "'" } else { "h" })
```

Comment:
This is reversed, in normal mode it should still print `'` and not `h`.

Also: just out of curiosity, what's the performance characteristic of `write_str` vs `write_char`? Even if it doesn't get optimized out it's probably nothing major though.

Reply:
Hmm, I think we should default to `h` rather than `'`.

Reply:
We can certainly do that, but it would break backwards compatibility in some sense. What would be the reason to do so?

Reply:
`'` sucks, it breaks shell escaping and is harder to read. I'm okay with the compatibility breakage.

Alternately we could make `{#}` output the `h` form like was suggested in the `Decimal` PR

Reply:
> Alternately we could make {#} output the h form like was suggested in the Decimal PR

That's what I wanted. But I'm ok with `{}` outputting `h` and `{#}` outputting `'` too if you consider our current behavior of always outputting `'` bad.

Reply:
> Alternately we could make {#} output the h form like was suggested in the Decimal PR

This is what happens now with the present version of the code:
```rust
                f.write_str(if f.alternate() { "'" } else { "h" })
```

So do I need to change this?

Reply:
The current version (`f.write_str(if f.alternate() { "'" } else { "h" })`) is what  @apoelstra wants afaik, but breaks compatibility. I'm fine with that as long as it happens consciously and not accidentally.